js/src/api/transport/ws/ws.js

```diff
@@ -81,13 +79,23 @@ export default class Ws extends JsonRpcBase {
       this._ws = null;
       this._sessionHash = null;
     }
-
     this._connecting = true;
     this._connected = false;
     this._lastError = null;
 
-    this._sessionHash = sha3;
-    this._ws = new WebSocket(this._url, hash);
+    // rpc secure API
+    if (this._token) {
+      const time = parseInt(new Date().getTime() / 1000, 10);
+      const sha3 = keccak_256(`${this._token}:${time}`);
+      const hash = `${sha3}_${time}`;
+
+      this._sessionHash = sha3;
+      this._ws = new WebSocket(this._url, hash);
+    // pubsub API
```

Comment:
Technically both secure and non-secure APIs suport pubsub, so I think the comment should rather say "non-secure API"